### ISSUE TITLE
Unable to set parameters on the selector within a SelectorProxy

### ISSUE THREAD

**Issue by tyers** [jpmml/sklearn2pmml#156]
Hi Villu,
I'm looking at implementing feature selection within my modelling pipelines, and can see from the jpmml-sklearn readme that this would need to be wrapped in a SelectorProxy.

Constructing the pipeline with fixed hyper-parameters is fine, however it is not possible to update the parameters of the wrapped selector (eg the k param of selectkbest) using pipeline.set_params or indeed adjust these settings as part of a round of hyper-parameter tuning; grid search etc.

When attempting to do this an error like this thrown: 
"ValueError: Invalid parameter selector for estimator SelectorProxy. Check the list of available parameters with `pipeline.get_params().keys()`." 
seems that the _ on the end of the selector_ parameter is tripping up that functions ability to traverse down through the pipeline to set any attributes correctly.

**Comment by vruusmann** [comment]
Solving this issue would require deeper understanding of Scikit-Learn library internals. Namely, the `SelectorProxy` class (and its "sibling" the `EstimatorProxy` class) should implement some parameter proxying methods to make everything work. Or, perhaps they should inherit from some additional Scikit-Learn classes (in addition to the default `BaseEstimator` class).

TLDR: Try to figure out the necessary Python code change yourself. Once you've got it, I'd be happy to integrate it into the SkLearn2PMML codebase.

**Comment by tyers** [comment]
Let me have a poke around - was a few weeks ago that i was looking at this last; i just forgot to raise the issue at the time.
will update when i have a clearer view but i think when i looked last time it was as simple as dropping the "_" from "selector_" - seems when scikit is tokenising the parameter using "__" to split the presence of the "_" on the end of the parameter causes the issue (guess this is due to python convention privacy being denoted by leading underscores?)

**Comment by vruusmann** [comment]
A quick workaround:
1. Use the regular `sklearn.pipeline.Pipeline` class, and regular (ie. unwrapped) selector classes.
2. Fit the pipeline.
3. Convert the fitted `Pipeline` to `PMMLPipeline` using the `sklearn2pmml.make_pmml_pipeline` utility function - will filter pipeline steps and wrap all fitted selectors into `SelectorProxy` for conversion purposes.
4. Convert the `PMMLPipeline` to a PMML file.

**Comment by tyers** [comment]
didn't realise i could do that! sounds exactly what i'm looking to do anyway - i thought i had to manually wrap everything when constructing the pipeline.
will have a look in to the error regardless.
